packages/netlify/src/image-service.ts

```diff
@@ -17,9 +17,18 @@ const service: ExternalImageService = {
 	getURL(options) {
 		const query = new URLSearchParams();
 
-		const fileSrc = isESMImportedImage(options.src)
+		let fileSrc = isESMImportedImage(options.src)
 			? removeLeadingForwardSlash(options.src.src)
 			: options.src;
+		const site = import.meta.env.SITE;
+		if (URL.canParse(fileSrc)) {
+			const url = new URL(fileSrc);
+			if (import.meta.env.NETLIFY_LOCAL && url.hostname === 'localhost') {
+				fileSrc = fileSrc.substring(url.origin.length);
+			} else if (site && fileSrc.startsWith(site)) {
+				fileSrc = fileSrc.substring(site.length);
+			}
+		}
```

Comment:
Do you have an example where it's plausible for someone to write image paths with the SITE (which is usually the prod site url), or `localhost`, which theoretically would've only worked locally?

If this is more of the URL being mismanaged and passed in this way, maybe it makes more sense to fix on the consumer side?

Reply:
The examples I've seen use `new URL(src, Astro.url.origin).href`, and they do it because `inferSize` needs a full URL. Maybe we should allow inferSize to work with local paths too.